### DISCUSSION TITLE
Pull through registry, guard against internet outage (registry unreachable)

### DISCUSSION THREAD

**Discussion by Ulrar** [siderolabs/talos#8059] [Q&A]
Hi,

I'm running Talos at home to run external images (mostly from the docker hub, some gcr), and would like to plan for internet outages. I'm already using the IfNotPresent policy on everything, but pods can be scheduled on nodes that don't have the image and if the internet is out, it won't start.

I've looked at dragonfly2, but it requires a sql database which I guess means a pvc. Given how unreliable my linstor has been, I'd probably just be making it worse going with that, but that's a different issue.
Dragonfly can be configured in a few different ways, I believe for Talos it'd be containerd with a config v2, and I guess let it inject config_path ? I'm not all that clear on how that works TBH, would that cause any issues with Talos ?

Kraken seems able to distribute images locally without one, but is abandoned. I haven't seen any successor, I may have missed it.

Any suggestions, tips or other projects to recommend ?

Thank you

**Comment by smira** [comment]
You can run a pull through registry, either a simple one using upstream `registry` image, or Gitea/Harbor to cache images for your cluster. Any in-cluster caching is usually subject to image GC, so it's not a solution to allow offline access.

**Reply by Ulrar** [reply]
I had no idea Gitea could do container registry now, that's neat since I already use it as the source for argocd. But looking at it they don't seem to have pull through support yet, unfortunately.
Harbor comes up a lot but that also requires a psql database which will mean a pvc.

I wish there was a way to get the images to be pulled on all nodes instead of just the one, that'd be the ideal solution really. Oh well, maybe I'll wait and see what Gitea comes out with and I'll try to replace linstor with something stable in the meantime.
Thanks

**Reply by smira** [reply]
you can use `talosctl image pull` to pre-pull images, but still kubelet can do image GC on its own
